src/resmom/start_exec.c

```diff
@@ -3566,7 +3566,7 @@ start_process(task *ptask, char **argv, char **envp, bool nodemux)
 	hook_output.progname = &progname;
 	CLEAR_HEAD(argv_list);
 	hook_output.argv = &argv_list;
-	hook_output.env = &env_str;
+	hook_output.env = NULL;
```

Comment:
I believe this will solve your immediate problem, but the feature in execjob_launch hook where one can modify the environment variable of the job will break. Give me some time to give you a possible solution. I've seen a fix to this and need to test it first.